interface/network_interface.cpp

```diff
@@ -0,0 +1,169 @@
+#include <cstring>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <net/if.h>
+#include <arpa/inet.h>
+#include <unistd.h>
+
+#include <sys/ioctl.h>
+
+#include "network_interface.h"
+#include "ilogger.h"
+#include "interface_addresses.h"
+
+NetworkInterface::NetworkInterface(const std::string& name, std::shared_ptr<ILogger> logger) :
+        m_logger(logger),
+        m_name(name)
+{
+    m_fd = socket(AF_INET, SOCK_DGRAM, 0);
+}
+
+NetworkInterface::~NetworkInterface()
+{
+    if (m_fd >= 0)
+    {
+        close(m_fd);
+    }
+}
+
+bool NetworkInterface::AddIpAddress(const std::string& ipAddress, const std::string& netmask) const
+{
+    if (m_fd < 0)
+    {
+        perror("socket");
+        LOG_ERROR(m_logger, "socket fd is negative: " + std::to_string(m_fd));
+        return false;
```

Comment:
Relevant for multiple places - this log is kinda weird - 
* socket always returns specifically -1 on failure, so there's no need to to std::to_string to m_fd and only write -1
https://man7.org/linux/man-pages/man2/socket.2.html
![image](https://github.com/ottopia-tech/keepalived-exporter/assets/79903220/61b7a326-0989-4f9c-892a-afd1817d0113)
* since you already printed perror with "socket" (that also prints the errno), what information does this log  give me? Is this only dump the log to the file? this seems inefficient, and you should probably just use OTTO_LOG with the errno.
* why not change the filedescriptors of std::cout\std::err to output to the log file? where does perror print it's info to? I assume the standard output location since I didn't see any change for it, some some data will be lost for you \ you need to output everything twice when error